server/controllers/auth_controller.js

```diff
@@ -33,18 +34,27 @@ const postRegister = (req, res, next) => {
 
   User.findOne({ username: props.username })
     .then(user => {
-      if (user) return next(createError({
-        status: CONFLICT,
-        message: 'Username already exists'
-      }))
+      if (user){
+        return res.status(CONFLICT).json({
```

Comment:
The error middleware is meant to handle error responses. Rather than returning a response directly, only need to call `next()` and pass it an error. To trigger a specific status error handler, use the `createError()` function and define its status.  See `/server/middlewar/error_middleware.js` ;)